cmake/findDependencies.cmake

```diff
@@ -1,6 +1,8 @@
 if (BUILD_GUI)
-    set(GUI_QT_COMPONENTS Core Gui Widgets PrintSupport)
-    find_package(Qt5 COMPONENTS ${GUI_QT_COMPONENTS})
+    find_package(Qt5Core)
```

Comment:
Why have you got other programming opinions than [Marcus D. Hanwell](https://blog.kitware.com/cmake-finding-qt5-the-right-way/ "CMake: Finding Qt 5 the “Right Way” (2016-10-21)")?

Reply:
Because Hanwell's code does not work in new CMake.

Hanwell's post is two years old.

Reply:
* Do you interpret a bit of information in the wrong direction?
* Would you like to recheck the involved software versions?
* I am using the command “`find_package(Qt5 COMPONENTS Core Widgets PrintSupport Sql REQUIRED)`” based on the software “[CMake 3.12.0-2.1](https://cmake.org/cmake/help/v3.13/command/find_package.html#basic-signature-and-module-mode "Description for find_package()")” and “[Qt 5.11.2-1.1](https://doc.qt.io/qt-5/cmake-manual.html "Qt CMake manual")” without build error messages for my recent development experiments.

Reply:
>     * Do you interpret a bit of information in the wrong direction?

Looks like you have. Old version has two lines of code that didn't work, my version has one line line of code that works without tuning the build system at all.

>     * Would you like to recheck the involved software versions?
 DEBUG util.py:439:   cmake                   x86_64 3.11.2-1.fc28                       build 7.7 M
DEBUG util.py:439:   qt5-devel               noarch 5.10.1-1.fc28                       build 9.9 k

>     * I am using the command “`find_package(Qt5 COMPONENTS Core Widgets PrintSupport Sql REQUIRED)`” based on the software “[CMake 3.12.0-2.1](https://cmake.org/cmake/help/v3.13/command/find_package.html#basic-signature-and-module-mode)” and “[Qt 5.11.2-1.1](https://doc.qt.io/qt-5/cmake-manual.html)” without build error messages for my recent development experiments.

This literally holds no information. You're presumably not using a vanilla system.

Picking a fight over A SINGLE LINE OF CODE just makes no sense. Does my version not work for you? Open up a ticket.

If you want to see for yourself that the old version did not work, install Fedora and do it.

I have no time for this sort of time waste.

Reply:
>Old version has two lines of code that didn't work,

I am curious to clarify your different observations from other run time environments a bit more.
Should it usually work if the needed software dependencies were fulfilled?


>my version has one line line of code

You split a list of components in a CMake variable into separate calls of the command “find_package”, didn't you?
Will a single function call become nicer in the script for Cppcheck's graphical user interface?


>cmake x86_64 3.11.2-1.fc28

The package source from Fedora 28 should be sufficient while I can share experiences from openSUSE Tumbleweed.


>This literally holds no information.

This command example indicates that I got the Qt SQL support working for my development system.


>Open up a ticket.

[Another clarification is evolving](https://github.com/danmar/cppcheck/pull/1428 "Complete CMake commands for integration of Qt software in Cppcheck's graphical user interface") for these build scripts.